packages/remix-dev/cli/migrate/migrations/replace-remix-imports/index.ts

```diff
@@ -4,41 +4,135 @@ import { join } from "path";
 import type { PackageJson } from "type-fest";
 
 import * as jscodeshift from "../../jscodeshift";
-import { cleanupPackageJson } from "./cleanupPackageJson";
-import { getTransformOptions } from "./getTransformOptions";
+import {
+  findRemixDependencies,
+  getTransformOptions,
+} from "./getTransformOptions";
 import type { Options } from "./transform";
+import { isAdapter, isRuntime } from "./transform";
 import type { MigrationFunction } from "../../types";
 import { readConfig } from "../../../../config";
+import { hint } from "../../../../logging";
+import * as colors from "../../../../colors";
 
 const transformPath = join(__dirname, "transform");
 
+function* getTasks({
+  packageJson,
+  runtime,
+  adapter,
+}: {
+  packageJson: PackageJson;
+  runtime: string;
+  adapter?: string;
+}): Generator<string> {
+  let commandTask = (message: string, command: string) =>
+    `${message}:\n\t${colors.hint(command)}`;
+  let remixRunPrefix = (dep: string) => "@remix-run/" + dep;
+  let remixDeps = findRemixDependencies(packageJson.dependencies);
+
+  // runtime not in deps
+  if (!remixDeps.includes(runtime)) {
+    yield commandTask(
+      `Install \`@remix-run/${runtime}\` as a dependency`,
+      `npm install @remix-run/${runtime}`
+    );
+  }
+
+  // other runtimes in deps
+  let otherRuntimes = remixDeps
+    .filter(isRuntime)
+    .filter((dep) => dep !== runtime);
+  if (otherRuntimes.length > 0) {
+    yield commandTask(
+      "Uninstall all unused runtimes",
+      `npm uninstall ${otherRuntimes.map(remixRunPrefix).join(" ")}`
+    );
+  }
+
+  // adapter not in deps
+  if (adapter && !remixDeps.includes(adapter)) {
+    yield commandTask(
+      `Install \`@remix-run/${adapter}\` as a dependency`,
+      `npm install @remix-run/${adapter}`
+    );
+  }
+
+  // other adapters in deps
+  let otherAdapters = remixDeps
+    .filter(isAdapter)
+    .filter((dep) => dep !== adapter);
+  if (otherAdapters.length > 0) {
+    yield commandTask(
+      "Uninstall all unused adapters",
+      `npm uninstall ${otherAdapters.map(remixRunPrefix).join(" ")}`
+    );
+  }
```

Comment:
This will never be shown, as `resolveAdapter` will fail when multiple adapters are present